### PR TITLE
feat: Ability to change branch for all selected applications during bulk build from Application Groups

### PR DESCRIPTION
**Pull request by Ashish-devtron**
# Description
Ability to change branch for all selected applications during bulk build from Application Groups

Fixes https://github.com/devtron-labs/devtron/issues/3887

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/ApplicationGroup/Details/TriggerView/EnvTriggerView.tsx [hunk 1]
```diff
@@ -194,6 +212,7 @@ export default function EnvTriggerView({ filteredAppIds, isVirtualEnv }: AppGrou
                     }),
                 )
             }
+            processWorkflow(_workflows)
```
Comment (vivek-devtron) [inline src/components/ApplicationGroup/Details/TriggerView/EnvTriggerView.tsx:215]: Rename it to preserveSelection

File: src/components/ApplicationGroup/Details/TriggerView/EnvTriggerView.tsx [hunk 2]
```diff
@@ -1032,6 +1051,54 @@ export default function EnvTriggerView({ filteredAppIds, isVirtualEnv }: AppGrou
         setFilteredWorkflows(_workflows)
     }
 
+    const getBranchChangeStatus = (statusText: string): BulkResponseStatus => {
```
Comment (vivek-devtron) [inline src/components/ApplicationGroup/Details/TriggerView/EnvTriggerView.tsx:1054]: Move it to utils or cost

File: src/components/ApplicationGroup/Details/TriggerView/SourceUpdateResponseModal.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+                <div
+                    className="dc__position-sticky dc__top-0 bcn-0 dc__border-bottom response-row dc__border-bottom pt-24 pb-8 dc__uppercase"
+                    style={{ zIndex: 1 }}
+                >
+                    <div className="fs-12 fw-6 cn-7 ">Application</div>
+                    <div className="fs-12 fw-6 cn-7 ">Branch Change status</div>
+                    <div className="fs-12 fw-6 cn-7 ">Message</div>
+                </div>
+                {responseList
+                    .sort((a, b) => sortCallback('appName', a, b))
```
Comment (vivek-devtron) [inline src/components/ApplicationGroup/Details/TriggerView/SourceUpdateResponseModal.tsx:21]: Please move the sorting logic in base

File: src/components/ApplicationGroup/Details/TriggerView/EnvTriggerView.tsx (hunk 2 above)
Comment (Ashish-hacker-0) [reply src/components/ApplicationGroup/Details/TriggerView/EnvTriggerView.tsx:1054]: Done

File: src/components/ApplicationGroup/Details/TriggerView/EnvTriggerView.tsx (hunk 1 above)
Comment (Ashish-devtron) [reply src/components/ApplicationGroup/Details/TriggerView/EnvTriggerView.tsx:215]: Renamed.

File: src/components/ApplicationGroup/Details/TriggerView/EnvTriggerView.tsx (hunk 2 above)
Comment (Ashish-devtron) [reply src/components/ApplicationGroup/Details/TriggerView/EnvTriggerView.tsx:1054]: Moved.

**Review by vivek-devtron — APPROVED** [review]

File: src/components/ApplicationGroup/Details/TriggerView/SourceUpdateResponseModal.tsx (hunk 3 above)
Comment (Ashish-devtron) [reply src/components/ApplicationGroup/Details/TriggerView/SourceUpdateResponseModal.tsx]: resolved

**Review by vivek-devtron — DISMISSED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Review by vivek-devtron — DISMISSED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**